ordering-keys-prober/src/main/java/com/google/cloud/pubsub/prober/README.md

```diff
@@ -0,0 +1,69 @@
+### Introduction
+
+The prober is designed to demonstrate effective use of Cloud Pub/Sub's
+[ordered delivery](https://cloud.google.com/pubsub/docs/ordering).
+
+### Building
+
+These instructions assume you are using [Maven](https://maven.apache.org/).
+
+1.  If you want to build the connector from head, clone the repository, ensuring
+    to do so recursively to pick up submodules:
+
+    `git clone https://github.com/GoogleCloudPlatform/pubsub`
+
+2.  Unzip the source code if downloaded from the release version.
+
+3.  Go into the ordering-keys-prober directory in the cloned repo.
+
+4.  Make the jar that contains the connector:
+
+    `mvn package`
+
+The resulting jar is at target/pubsub-ordered-prober.jar.
+
+### Running the Prober
+
+After building the prober, use the following command to run it:
+
+`java -cp target/pubsub-ordered-prober.jar com.google.cloud.pubsub.prober.ProberStarter --project <project ID>`
+
+When the prober starts up, it does the following:
+
+1.  Deletes the topic and subscription provided if they already exist.
+
+2.  Creates the topic and subscription provided.
+
+3.  Sets up subscribers to receive messages.
+
+4.  Continuously publishes messages.
+
+### Prober Properties
+
+The prober has many properties that can be set to test different scenario. The
+following table details all of the command-line options that can be set.
+
+| Property                                 | Value Type | Default | Description |
+|------------------------------------------|----------------------------|------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| project                                  | String                     | REQUIRED (No default)                    | The project ID in which to create the topic and subscription. |
+| endpoint                                 | String                     | "pubsub.googleapis.com:443"              | The Cloud Pub/Sub endpoint to send requests to. |
+| topic_name                               | String                     | "cloud-pubsub-client-library-prober"     | The name of the topic to create and publish messages to. |
+| subscription_name                        | String                     | "cloud-pubsub-client-library-prober-sub" | The name of the subscription to create and to receive messages from. |
+| subscription_type                        | `STREAMING_PULL` or `PULL` | `STREAMING_PULL`                         | The type of subscriber to create. |
```

Comment:
Maybe we should:
1. add links to streaming pull/pull documentation so readers can understand what the different subscriber types are, and
2. discuss a little how pull vs streaming pull would affect ordered behavior w.r.t affinity for example.

Reply:
Added the link. However, the second point will be discussed in the blog post.